src/app/components/dynamic-hub/dynamic-hub.component.ts

```diff
@@ -125,91 +125,212 @@ export class DynamicHubComponent implements OnInit {
                 }
             }
         }
+    }
+
+    public add() {
+        if (this.authentMode === 'iam' && this.connected) {
+            const iamHeader = {
+                Authorization: 'Bearer ' + this.arlasIamService.getAccessToken()
+            };
+            const options = { headers: iamHeader };
+            const action: HubAction = {
+                type: HubActionEnum.CREATE,
+                orgs: this.orgs,
+                options: options
+            };
+            const dialogRef = this.dialog.open(HubActionModalComponent, {
+                disableClose: true,
+                data: action
+            });
+            dialogRef.afterClosed()
+                .pipe(filter(result => result !== false))
+                .subscribe(result => {
+                    this.fetchCards();
+                    let url = this.arlasSettingsService.getArlasBuilderUrl().concat('/load/').concat(result[0]);
+                    if (!!result[1]) {
+                        url = url.concat(`?org=${result[1]}`);
+                    }
+                    const win = window.open(url, '_blank');
+                    win.focus();
+                });
 
+        } else {
+            const action: ConfigAction = {
```

Comment:
So if not logged with IAM or not logged at all, uses a ConfigAction and not a HubAction, because there is no notion of organisation, isn't it ?

Reply:
yes totally